### DISCUSSION TITLE
How do I calculate perplexity?

### DISCUSSION THREAD

**Discussion by ShrimpLeopold** [facebookresearch/ParlAI#4268] [Q&A]
Hi there! I need to check if my dialogue model's output is a valid English sentence. As far I'm concerned, it can be done by calculating perplexity of model's output. 
I use transformer/generator model and create a agent with create_agent() function passing a model file into it.
I wonder whether I should create a Torch Generator Agent cause it has the forward method (the prolem is that I don't understand how to create it) or is there another way?

**Comment by stephenroller** [comment]
Easiest way is to follow our teacher tutorial to create a dataset containing the phrases you wish to evaluate, and use eval_model with that dataset.

**Reply by ShrimpLeopold** [reply]
Sorry, but the point is to measure PPL of output to new, unknown phrases
